src/TaskManager/API/Models/TaskExecutionStats.cs

```diff
@@ -0,0 +1,127 @@
+
+        /// <summary>
+        /// Gets or sets Db version.
+        /// </summary>
+        [JsonConverter(typeof(DocumentVersionConvert)), BsonSerializer(typeof(DocumentVersionConverBson))]
+        public DocumentVersion Version { get; set; } = new DocumentVersion(1, 0, 0);
+
+        /// <summary>
+        /// the correlationId of the event
+        /// </summary>
+        [JsonProperty(PropertyName = "correlation_id")]
+        [Required]
+        public string CorrelationId { get; set; }
+
+        /// <summary>
+        /// the workflow Instance that triggered the event
+        /// </summary>
+        [JsonProperty(PropertyName = "workflow_instance_id")]
+        [Required]
+        public string WorkflowInstanceId { get; set; }
+
+        /// <summary>
+        /// This execution ID
+        /// </summary>
+        [JsonProperty(PropertyName = "execution_id")]
+        [Required]
+        public string ExecutionId { get; set; }
+
+        /// <summary>
+        /// The event Task ID
+        /// </summary>
+        [Required]
+        [JsonProperty(PropertyName = "task_id")]
+        public string TaskId { get; set; }
+
+        /// <summary>
+        /// the messageId of the message
+        /// </summary>
+        [JsonProperty(PropertyName = "message_id")]
+        [Required]
+        public string MessageId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the date time that the task started with the plug-in.
+        /// </summary>
+        [JsonProperty(PropertyName = "started")]
+        public DateTime Started { get; set; }
+
+        /// <summary>
+        /// Gets or sets the date time that the task started with the plug-in.
+        /// </summary>
+        [JsonProperty(PropertyName = "lastUpdated")]
+        public DateTime LastUpdated { get; set; }
+
+        /// <summary>
+        /// Gets or sets the date time that the task started with the plug-in.
+        /// </summary>
+        [JsonProperty(PropertyName = "completedAt")]
+        public DateTime CompletedAt { get; set; }
+
+        /// <summary>
+        /// Gets or sets the date time that the task started with the plug-in.
+        /// </summary>
+        [JsonProperty(PropertyName = "executionTimeSeconds")]
+        public double ExecutionTimeSeconds { get; set; }
+
+        /// <summary>
+        /// Gets or sets the date time that the task started with the plug-in.
+        /// </summary>
+        [JsonProperty(PropertyName = "status")]
+        public string Status { get; set; } = TaskStatus.Created.ToString();
+
+        public TaskExecutionStats(TaskDispatchEventInfo dispatchInfo)
+        {
+            CorrelationId = dispatchInfo.Event.CorrelationId;
+            WorkflowInstanceId = dispatchInfo.Event.WorkflowInstanceId;
+            ExecutionId = dispatchInfo.Event.ExecutionId;
+            TaskId = dispatchInfo.Event.TaskId;
+            Started = dispatchInfo.Started;
+            Status = dispatchInfo.Event.Status.ToString();
```

Comment:
think you need to null check this for your tests.

Reply:
Guards now added. :)